src/window.rs

```diff
@@ -251,6 +251,45 @@ impl WindowWrapper {
     }
 }
 
+#[derive(Clone)]
+struct WindowSettings {
+    refresh_rate: u64,
+    no_idle: bool,
+}
+
+fn parse_changed_setting(name: &str, value: Option<Value>) -> Value {
+    let mut settings = SETTINGS.get::<WindowSettings>();
+    match name {
```

Comment:
Minor, but you could probably simplify this by passing a lambda directly to the add_listener function which does the parsing for each setting directly. That way you don't have to do the string comparison, and don't have to have the match statement.

Further, if you do that, I think you could probably remove the name argument to the listeners type completely. add_listener specifies the name of the variable in the hookup, so its sorta redundant to have the name in the callback as well.